datalad_next/tree.py

```diff
@@ -0,0 +1,793 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See LICENSE file distributed along with the datalad_osf package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""'tree'-like command for visualizing dataset hierarchies"""
+
+__docformat__ = "numpy"
+
+import logging
+from functools import wraps, lru_cache
+from pathlib import Path
+
+from datalad.interface.base import (
+    Interface,
+    build_doc,
+)
+from datalad.support.param import Parameter
+from datalad.distribution.dataset import (
+    datasetmethod,
+    require_dataset,
+)
+from datalad.interface.results import (
+    get_status_dict,
+)
+from datalad.interface.utils import (
+    eval_results,
+)
+from datalad.support.constraints import (
+    EnsureNone,
+    EnsureStr, EnsureInt, EnsureRange,
+)
+from datalad.support import ansi_colors
+
+lgr = logging.getLogger('datalad.local.tree')
+
+
+@build_doc
+class TreeCommand(Interface):
```

Comment:
True. When trying to follow the convention, I got naming exhaustion syndrome from trying to come up with a synonym for `Tree` to name the actual class for tree generation :-). I thought it was fine since we can explicitly set the class name in the command suite definition. Or would you rather keep the naming convention for the sake of consistency?

Reply:
Nah, all good. I just wanted to mention it.